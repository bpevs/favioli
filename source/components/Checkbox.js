import React from "react"


export default () => <div className="checkbox">
  <label className="help">Flag Replaced Favicons</label>
  <input id="flag" name="flag" type="checkbox" />
  <div className="checkmark" />
</div>
