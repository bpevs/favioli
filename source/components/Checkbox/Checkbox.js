import React from "react"


function opt (context, func) {
  if (typeof func === "function") return func.bind(context)
  return () => {}
}

export class Checkbox extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      focused: false
    }
  }

  onBlur() {
    this.setState(
      { focused: false },
      opt(this, this.props.onBlur)
    )
  }

  onFocus() {
    this.setState(
      { focused: true },
      opt(this, this.props.onFocus)
    )
  }

  render() {
    const {
      checked = false,
      name,
      onChange,
    } = this.props

    const focusClass = this.state.focused ? " focused": ""

    return (
      <div className={"checkbox" + focusClass}>
        <label className="help">Flag Replaced Favicons</label>
        <input id="flag" name={name} type="checkbox"
          checked={checked}
          onBlur={this.onBlur.bind(this)}
          onChange={e => {
            onChange({ [e.target.name]: e.target.checked })
          }}
          onFocus={this.onFocus.bind(this)}
          tabIndex={0}
        />
        <div className="checkmark" />
      </div>
    )
  }
}
