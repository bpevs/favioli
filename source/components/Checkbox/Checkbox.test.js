import { mount, shallow } from "enzyme"
import * as React from "react"
import { Checkbox } from "./Checkbox"


test("Should Render Checkbox", () => {
  const wrapper = shallow(<Checkbox
    checked={true}
    name="my-checkbox"
    onChange={jest.fn()}
  />)

  expect(wrapper).toMatchSnapshot()
});

test("Should Render Checkbox unchecked", () => {
  const wrapper = shallow(<Checkbox
    checked={false}
    name="my-checkbox1"
    onChange={jest.fn()}
  />)

  expect(wrapper).toMatchSnapshot()
});

test("Should run onChange on change", () => {
  const onChanged = jest.fn()
  const wrapper = mount(<Checkbox
    name="my-checkbox3"
    onChange={onChanged}
  />)

  wrapper.find("input").simulate("change")
  expect(wrapper).toMatchSnapshot()
  expect(onChanged).toBeCalled()
});
