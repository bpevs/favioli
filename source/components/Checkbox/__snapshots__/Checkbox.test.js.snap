// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Checkbox 1`] = `
<div
  className="checkbox"
>
  <label
    className="help"
  >
    Flag Replaced Favicons
  </label>
  <input
    checked={true}
    id="flag"
    name="my-checkbox"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    tabIndex={0}
    type="checkbox"
  />
  <div
    className="checkmark"
  />
</div>
`;

exports[`Should Render Checkbox unchecked 1`] = `
<div
  className="checkbox"
>
  <label
    className="help"
  >
    Flag Replaced Favicons
  </label>
  <input
    checked={false}
    id="flag"
    name="my-checkbox1"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    tabIndex={0}
    type="checkbox"
  />
  <div
    className="checkmark"
  />
</div>
`;

exports[`Should run onChange on change 1`] = `
<Checkbox
  name="my-checkbox3"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "my-checkbox3": false,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="checkbox"
  >
    <label
      className="help"
    >
      Flag Replaced Favicons
    </label>
    <input
      checked={false}
      id="flag"
      name="my-checkbox3"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      tabIndex={0}
      type="checkbox"
    />
    <div
      className="checkmark"
    />
  </div>
</Checkbox>
`;
